scripts/cast-on-tenderly/index.js

```diff
@@ -0,0 +1,110 @@
+    throw new Error('Please provide address of the spell, e.g.: `node index.js 0x...`');
+}
+
+const makeTenderlyApiRequest = async function (path) {
+    const API_BASE = `https://api.tenderly.co/api/v1/account/${process.env.TENDERLY_USER}/project/${process.env.TENDERLY_PROJECT}`;
+    return await axios.post(
+        `${API_BASE}${path}`,
+        { network_id: NETWORK_ID },
+        { headers: { 'X-Access-Key': process.env.TENDERLY_ACCESS_KEY } }
+    );
+};
+
+const createTenderlyFork = async function () {
+    const response = await makeTenderlyApiRequest('/fork');
+    const forkId = response.data.simulation_fork.id;
+    const rpcUrl = `https://rpc.tenderly.co/fork/${forkId}`;
+    const forkUrl = `https://dashboard.tenderly.co/${process.env.TENDERLY_USER}/${process.env.TENDERLY_PROJECT}/fork/${forkId}`;
+    return { forkId, forkUrl, rpcUrl };
+};
+
+const publishTenderlyTransaction = async function (forkId, transactionId) {
+    await makeTenderlyApiRequest(`/fork/${forkId}/transaction/${transactionId}/share`);
+    return `https://dashboard.tenderly.co/shared/fork/simulation/${transactionId}`;
+};
+
+const runSpell = async function () {
+    const { forkId, forkUrl, rpcUrl } = await createTenderlyFork();
+    console.info('private tenderly fork is created', forkUrl);
+
+    const provider = new ethers.providers.JsonRpcProvider(rpcUrl);
+    const signer = provider.getSigner();
+
+    console.info('fetching the chief address from chainlog...');
+    const chainlog = new Contract(
+        CHAINLOG_ADDRESS,
+        ['function getAddress(bytes32) external view returns (address)'],
+        signer
+    );
+    const chiefAddress = await chainlog.getAddress(ethers.utils.formatBytes32String('MCD_ADM'));
+
+    console.info('overwriting the hat...');
+    await provider.send('tenderly_setStorageAt', [
+        chiefAddress,
+        ethers.utils.hexZeroPad(ethers.utils.hexValue(CHIEF_HAT_SLOT), 32),
+        ethers.utils.hexZeroPad(SPELL_ADDRESS, 32),
+    ]);
+
+    console.info('checking the hat...');
+    const chief = new Contract(chiefAddress, ['function hat() external view returns (address)'], signer);
+    const hatAddress = await chief.hat();
+    if (hatAddress !== SPELL_ADDRESS) {
+        throw new Error('spell does not have the hat');
+    }
+
+    const spell = new Contract(
+        SPELL_ADDRESS,
+        ['function schedule() external', 'function cast() external', 'function eta() external view returns (uint256)'],
+        signer
+    );
+    console.info('scheduling spell on a fork...');
+    try {
+        const scheduleTx = await spell.schedule(DEFAULT_TRANSACTION_PARAMETERS);
+        await scheduleTx.wait();
+    } catch (error) {
+        console.warn('scheduling failed', error);
+    }
+
+    console.info('fetching timestamp when the spell will be castable...');
+    const eta = await spell.eta();
+
+    console.info(`warping the time to "${eta}"...`);
+    const currentUnixTimestamp = Date.now() / 1000;
```

Comment:
too late of a comment, but i think `parseInt` is due because it can result into float here and below we do `hexValue()` call on the division result

Reply:
Thanks for spotting it and noting it down here! 

Although I would suggest to use `Math.ceil()` over `parseInt()`. Because `parseInt()` is for [string parsing](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt), not for rounding 🙂

Reply:
(this was later fixed by https://github.com/makerdao/spells-goerli/pull/229)